pkg/multicluster/kubeconfig/kubeconfig_secret.go

```diff
@@ -31,22 +52,44 @@ func ToSecret(namespace string, cluster string, kc api.Config) (*kubev1.Secret,
 			Namespace: namespace,
 		},
 		Type: SecretType,
-		Data: map[string][]byte{Key: rawKubeConfig},
+		Data: map[string][]byte{secretKey: rawKubeConfig},
 	}, nil
 }
 
-// Config contains various KubeConfig formats for convenience.
-type Config struct {
-	ClientConfig clientcmd.ClientConfig
-	ApiConfig    *api.Config
-	RestConfig   *rest.Config
+// https://github.com/solo-io/service-mesh-hub/issues/590
+// If the user has a cert authority file set instead of the raw bytes in their kubeconfig, then
+// we'll fail later when the pods in-cluster try to read that file path.
+// We need to read the file right now, in a CLI context, and manually shuffle the bytes over to the CA data field
+func readCertAuthFileIfNecessary(cfg api.Config) error {
```

Comment:
i'd add an explanation here of why this is necessary (**we are reading in a kubeconfig stored on a user's local machine, which may not use inline certs**)

i'd maybe rename the function to `convertCertFilesToInline`